gui/src/screen/mod.rs

```diff
@@ -7,17 +7,37 @@ use std::sync::Arc;
 mod engine;
 
 #[cfg(not(target_os = "macos"))]
-pub type Default = self::engine::Vulkan;
+pub type DefaultApi = self::engine::Vulkan;
 
 #[cfg(target_os = "macos")]
-pub type Default = self::engine::Metal;
+pub type DefaultApi = self::engine::Metal;
 
 #[cfg(not(target_os = "macos"))]
-pub type ScreenError = self::engine::VulkanError;
+pub type Default = self::engine::VulkanScreen;
+
+#[cfg(target_os = "macos")]
+pub type Default = self::engine::MetalScreen;
+
+#[cfg(not(target_os = "macos"))]
+pub type ScreenError = self::engine::VulkanScreenError;
 
 #[cfg(target_os = "macos")]
 pub type ScreenError = self::engine::MetalError;
 
+pub trait GraphicsApi: Sized + 'static {
+    type PhysicalDevice: PhysicalDevice;
+
+    type InitError: Error;
+
+    fn init() -> Result<Self, Self::InitError>;
+
+    fn enumerate_physical_devices(&self) -> &[Self::PhysicalDevice];
+}
+
+pub trait PhysicalDevice: Sized {
+    fn name(&self) -> &str;
+}
+
```

Comment:
This and its implementations should be a top level module.

Reply:
Like src/graphics?

Reply:
Right.